cloudant-sync-datastore-core/src/main/java/com/cloudant/mazha/CouchClient.java

```diff
@@ -124,16 +201,65 @@ private InputStream executeToInputStream(HttpConnection connection) throws Couch
         }
     }
 
-    private <T> T executeToJsonObject(HttpConnection connection, Class<T> c) throws CouchException {
-        InputStream is = this.executeToInputStream(connection);
+    // execute HTTP task with retries:
+    // return an InputStream if successful or throw an exception
+    private InputStream executeWithRetry(final Callable<ExecuteResult> task) throws CouchException {
+        int retries = 10;
```

Comment:
Again I find the `retries` terminology slightly misleading. As coded these are `tries` or `attempts`, but `retries` suggests the while loop should be `>= 0` again.